### ISSUE TITLE
Can't run cosmic-ray on TinyDB

### ISSUE THREAD

**Issue by msiemens** [sixty-north/cosmic-ray#238]
This is a fun one. It's not possible to run CR on TinyDB. Doing so results in a false surviral rate of > 90 %. Turns out, CR imports TinyDB before running py.test (which of course makes sense) so py.test decides to use the version of TinyDB bundled with CR instead of the local source files.

It's fine with me to just close this as a funny wont-fix as I just have to rename the `tinydb` module on my file system to something else to get it working, just thought I'd share this 🙂

**Comment by abingham** [comment]
This is an interesting one. CR does so much "meta" stuff that I think issues like this will be around for a while. If you install CR into a virtual environment that already has TinyDB installed (i.e. the version of it that you want to test), does CR mutate that TinyDB? It shouldn't try to install a new one, I think, if it sees an existing version.

It might be best if CR just incorporates a version of TinyDB into itself so that client programs can use TinyDB without this kind of interference.

**Comment by msiemens** [comment]
Yeah, I was too lazy to use a virtualenv. There were basically two versions present, one in the current directory and one installed as a dependency. It seems like CR mutated the one in the current directory while py.test ran the test suite of the not mutated version installed as a dependency.

> It might be best if CR just incorporates a version of TinyDB into itself so that client programs can use TinyDB without this kind of interference.

I'm just fine with leaving everything as is. This is not so much a "please fix this" but more of a "hey, I found this weird, funny corner case" 🙂

**Comment by abingham** [comment]
If it's not blocking you, then I'm not going to do anything about this right now. But I'm going to keep this open to remind me to look into when I have some time.

**Comment by abingham** [comment]
We're moving over to sqlite soon, so I'm going to close this.
